exec/java-exec/src/test/java/org/apache/drill/TestCorrelation.java

```diff
@@ -58,4 +58,50 @@ public void testScalarAggAndFilterCorrelatedSubquery() throws Exception {
       .run();
   }
 
+  @Test
+  public void testExistsScalarSubquery() throws Exception {
+    String query =
+        "SELECT employee_id\n" +
+        "FROM cp.`employee.json`\n" +
+        "WHERE EXISTS\n" +
+        "    (SELECT *\n" +
+        "     FROM cp.`employee.json` cs2\n" +
+        "     )\n" +
+        "limit 1";
+
+    testBuilder()
+        .sqlQuery(query)
+        .unOrdered()
+        .baselineColumns("employee_id")
+        .baselineValues(1L)
+        .go();
+  }
+
+  @Test
+  public void testSeveralExistsCorrelateSubquery() throws Exception {
```

Comment:
This query pattern reminds me of TPC-H query 21, although that one is doing EXISTS and NOT-EXISTS.  Currently, that test is marked Ignored [1].  Would this PR also handle that case ? 

[1] https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/TestTpchDistributed.java#L141

Reply:
That's good to know and thanks for enabling those tests.  Depending on who is testing this enhancement, you could let them know that the functional regression tests also need to be updated for higher scale factors.